src/lib.rs

```diff
@@ -608,6 +616,35 @@ impl<Types: IpfsTypes> Ipfs<Types> {
         rx.await?
     }
 
+    pub async fn bootstrap(&self) -> Result<(), Error> {
+        let (tx, rx) = oneshot_channel::<Result<FutureSubscription<(), String>, Error>>();
+
+        self.to_task.clone().send(IpfsEvent::Bootstrap(tx)).await?;
+
+        rx.await??.await?.map_err(|e| anyhow!(e))
+    }
+
+    pub async fn add_peer(&self, peer_id: PeerId, addr: Multiaddr) -> Result<(), Error> {
+        self.to_task
+            .clone()
+            .send(IpfsEvent::AddPeer(peer_id, addr))
+            .await?;
+
+        Ok(())
+    }
+
+    pub async fn get_closest_peers(&self) -> Result<(), Error> {
+        let self_peer = PeerId::from_public_key(self.identity().await?.0);
+        let (tx, rx) = oneshot_channel::<FutureSubscription<(), String>>();
+
+        self.to_task
+            .clone()
+            .send(IpfsEvent::GetClosestPeers(self_peer, tx))
+            .await?;
+
+        rx.await?.await?.map_err(|e| anyhow!(e))
+    }
```

Comment:
I already got confused, should this be the bootstrap or not?

Reply:
finding closest peers doesn't result in connecting to them at this moment; it's practical for testing DHT functionalities or doing related measurements, not unlike `bitswap_stats` that serves similar purposes

Reply:
How does this allow measuring? bitswap_stats is on the http API (can be used for monitoring) and is interesting when we can start caring about duplicate blocks and such.

Reply:
for measuring how fast we can find discover peers and in what scenarios, e.g. wrt the number of nodes and how they are connected (meshed/linked); I can remove that API, but this was my primary tool to verify that I had peer discovery working :stuck_out_tongue:

Reply:
Yeah maybe perhaps there could be a method comment to avoid anyone trying to use this for anything, or noting it's used in the test to produce log messages?

Reply:
I can just move it to `Node` that is only used in tests :ok_hand: